src/librustc_mir_build/hair/cx/expr.rs

```diff
@@ -728,10 +727,17 @@ fn convert_path_expr<'a, 'tcx>(
                     adt_def,
                     variant_index: adt_def.variant_index_with_ctor_id(def_id),
                     substs,
-                    user_ty: user_provided_type,
+                    user_ty,
                     fields: vec![],
                     base: None,
                 },
+                _ if ty.references_error() => {
+                    // Handle degenerate input without ICE (#67377).
+                    ExprKind::Literal {
+                        literal: ty::Const::zero_sized(cx.tcx, cx.tcx.types.err),
```

Comment:
Please use `tcx.consts.err`, doing it this way makes it slip past #71049, had I not come across this PR for a *completely unrelated* reason.

Reply:
Wait how are we even getting here?! MIR isn't built if `typeck_tables_of(def_id).is_tainted_by_errors` and that's where `ty` comes from.